language/move-core/types/src/language_storage.rs

```diff
@@ -79,6 +103,153 @@ impl<'a> From<&'a TypeTag> for SafeTypeTag<'a> {
     }
 }
 
+// u8 -> depth or nesting level
+struct TypeTagVisitor(u8);
+
+use serde::de::{
+    self, DeserializeSeed, Deserializer, EnumAccess, MapAccess, SeqAccess, VariantAccess, Visitor,
+};
+
+impl<'de> Visitor<'de> for TypeTagVisitor {
```

Comment:
I would be nice to see some commenting here, like

```
// Manual implementation of what serde would derive automatically, but with an injected bound. We are using this 
// here as currently no configurable type depth is supported by serde. See [link if there is something to see].
```
Or whatever makes sense. Otherwise, someone seeing this code here would just declare us as insane.

Also as you seem to have reverse engineered Serde, it would be nice you give some hints so others don't need to do the same as far as this is possible.

Reply:
this is how I keep my job...